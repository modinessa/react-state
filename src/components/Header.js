import React from "react";
import bigLogo from "../assets/your-logo-here.png"
import {useLocation} from "react-router-dom";

export function Header() {
	const location = useLocation();

	return location.pathname === '/' ? (
		<header className="app-section app-section--main-page app-section--image-peak">
		<img 
				src={bigLogo}
				alt="Logo icon" className="app-logo"
		/>
		<h1 className="app-title">
			Your Headline <br />
			Here
			</h1>
			<h2 className="app-subtitle">
				Lorem ipsum dolor sit amet, consectetur <br />
				adipiscing elit sed do eiusmod.
			</h2>
	</header>
	) : (
		<header className="app-section app-section--general app-section--image-peak">
		</header>
	)
}