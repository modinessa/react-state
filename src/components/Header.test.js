import {BrowserRouter} from 'react-router-dom'; 
import renderer from 'react-test-renderer';

import {Header} from './Header';

describe('Header', () => {
	it('renders full header for main page', () => {
		window.history.pushState({}, '', '/');

		const header = renderer.create(<BrowserRouter><Header/></BrowserRouter>);
		
		expect(header).toMatchSnapshot();
	});
	
	it('renders header for other than main page', () => {
		window.history.pushState({}, '', '/bla-bla-bla');

		const header = renderer.create(<BrowserRouter><Header/></BrowserRouter>);
		
		expect(header).toMatchSnapshot();
	});
});
