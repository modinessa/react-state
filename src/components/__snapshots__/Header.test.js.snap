// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders full header for main page 1`] = `
<header
  className="app-section app-section--main-page app-section--image-peak"
>
  <img
    alt="Logo icon"
    className="app-logo"
    src="your-logo-here.png"
  />
  <h1
    className="app-title"
  >
    Your Headline 
    <br />
    Here
  </h1>
  <h2
    className="app-subtitle"
  >
    Lorem ipsum dolor sit amet, consectetur 
    <br />
    adipiscing elit sed do eiusmod.
  </h2>
</header>
`;

exports[`Header renders header for other than main page 1`] = `
<header
  className="app-section app-section--general app-section--image-peak"
/>
`;
